packages/thirdweb/src/extensions/erc20/read/isERC20.ts

```diff
@@ -13,8 +22,19 @@
  * ```
  */
 export function isERC20(options: BaseTransactionOptions) {
-  return supportsInterface({
-    contract: options.contract,
-    interfaceId: "0x36372b07",
-  });
+  // there is no trustworthy way to check if a contract is ERC20 via ERC165, so we do this manually.
+  // see: https://github.com/OpenZeppelin/openzeppelin-contracts/issues/3575
+  // see: https://ethereum.org/en/developers/docs/standards/tokens/erc-20/
+
+  return allOf(
```

Comment:
feels like this should be the fallback if supportsInterface fails?  would still try to rely on it, its pretty widespread i think

Reply:
not for ERC20 (see the comments above)